### PR TITLE
Client 88. Admin -> user form enhancement (create/edit)

### PR DESCRIPTION
**Pull request by OlegSubik**
### Describe the background of your pull request

- Added yup, formik

- Enhanced user form with new fields 

- Changed the form's data handling.

### Additional context

Add any other context

### Governance 

- [ ] Documentation is added
- [ ] Test cases are added or updated
- [ ] I've read the contributing document https://github.com/cherrytwist/.github/blob/master/CONTRIBUTING.md
- [ ] I've read and understand the Code of Conduct https://github.com/cherrytwist/.github/blob/master/CODE_OF_CONDUCT.md
- [ ] I understand that any contributions or suggestions I made may make it into the actual code. I've read the License 
     https://github.com/cherrytwist/Coordination/blob/master/LICENSE 
     and the contributor license agreement: tba

### CODE REVIEW

File: src/components/Admin/UserFrom.tsx [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+  edit,
+  new,
+}
+interface UserProps {
+  users: Array<UserModel>;
+  editMode?: EditMode;
+  onSave?: (user: UserModel) => void;
+}
+
+export const UserFrom: FC<UserProps> = ({ users, editMode = EditMode.readOnly, onSave: _onSave }) => {
```
Comment (atstojanov) [inline src/components/Admin/UserFrom.tsx:33]: User**From** or User**Form**?

File: src/components/Admin/UserFrom.tsx [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+import { Alert, Button, Col, Form, FormControl } from 'react-bootstrap';
+import generator from 'generate-password';
+import { Formik, FieldArray } from 'formik';
+import * as yup from 'yup';
+/*lib imports end*/
+
+import InputWithCopy from './InputWithCopy';
+/*components imports end*/
+
+import gql from 'graphql-tag';
```
Comment (atstojanov) [inline src/components/Admin/UserFrom.tsx:12]: This is lib import. 
Please import from '@apollo/client';

`import { gql } from '@apollo/client';`

File: tsconfig.json [hunk 3]
```diff
@@ -17,12 +13,9 @@
     "resolveJsonModule": true,
     "isolatedModules": true,
     "noEmit": true,
+    "noImplicitAny": false,
```
Comment (atstojanov) [inline tsconfig.json:16]: Why?

Comment (OlegSubik) [reply tsconfig.json:16]: had a problem with setting adaptive form values `value[fieldName]`

File: src/components/Admin/UserFrom.tsx (hunk 1 above)
Comment (OlegSubik) [reply src/components/Admin/UserFrom.tsx:33]: yah, typo

**Review by techsmyth — APPROVED** [review]
Looks good! 

Some layout / visual items we can pick up as a whole next week when finalise (e.g. bounding box around refs)

**Pull request merged**